src/content/docs/fr/reference/api-reference.mdx

```diff
@@ -0,0 +1,1948 @@
+	url: string | undefined;
+	default: AstroComponent;
+}
+```
+
+#### Autres fichiers
+
+D'autres fichiers peuvent avoir des interfaces différentes, mais `Astro.glob()` accepte un générique TypeScript si vous savez exactement ce que contient un type de fichier non reconnu.
+
+```ts
+---
+interface CustomDataFile {
+  default: Record<string, any>;
+}
+const data = await Astro.glob<CustomDataFile>('../data/**/*.js');
+---
+```
+
+### `Astro.props`
+
+`Astro.props` est un objet contenant toutes les valeurs qui ont été transmises en tant qu'[attributs de composant](/fr/basics/astro-components/#props-de-composant). Les composants de mise en page pour les fichiers `.md` et `.mdx` reçoivent les valeurs de frontmatter comme props.
+
+```astro {3}
+---
+// src/components/Heading.astro
+const { title, date } = Astro.props;
+---
+<div>
+  <h1>{title}</h1>
+  <p>{date}</p>
+</div>
+```
+
+```astro /title=".+"/ /date=".+"/
+---
+// src/pages/index.astro
+import Heading from '../components/Heading.astro';
+---
+<Heading title="Mon tout premier article" date="09 Août 2022" />
+```
+
+<ReadMore>En savoir plus sur la façon dont les [Dispositions Markdown et MDX](/fr/guides/markdown-content/#frontmatter-layout) gèrent les props.</ReadMore>
+
+<ReadMore>Apprenez à ajouter des [définitions de type TypeScript pour vos props](/fr/guides/typescript/#type-componentprops).</ReadMore>
+
+### `Astro.params`
+
+`Astro.params` est un objet contenant les valeurs des segments de routes dynamiques correspondant à cette requête.
+
+Dans les versions statiques, il s'agira des `params` renvoyés par `getStaticPaths()` utilisés pour le pré-rendering [des routes dynamiques](/fr/guides/routing/#routes-dynamiques).
+
+Dans les versions SSR, il peut s'agir de n'importe quelle valeur correspondant aux segments de chemin dans le modèle de route dynamique.
+
+```astro title="src/pages/posts/[id].astro"
+---
+export function getStaticPaths() {
+  return [
+    { params: { id: '1' } },
+    { params: { id: '2' } },
+    { params: { id: '3' } }
+  ];
+}
+
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Voir aussi : [`params`](#params)
+
+### `Astro.request`
+
+`Astro.request` est un objet standard [Request](https://developer.mozilla.org/fr/docs/Web/API/Request). Il peut être utilisé pour obtenir le `url`, `headers`, `method`, et même le corps de la requête.
+
+```astro
+<p>Reçu une requête {Astro.request.method} depuis "{Astro.request.url}".</p>
+<p>En-têtes de requête reçus : <code>{JSON.stringify(Object.fromEntries(Astro.request.headers))}</code>
+```
+
+Voir aussi : [`Astro.url`](#astrourl)
+
+:::note
+Avec l'option par défaut `output : 'static', `Astro.request.url` ne contient pas de paramètres de recherche, comme `?foo=bar`, car il n'est pas possible de les déterminer à l'avance lors des constructions statiques. Cependant, en mode `output : 'server'`, `Astro.request.url` contient les paramètres de recherche car ils peuvent être déterminés à partir d'une requête du serveur.
```

Comment:
```suggestion
Avec l'option par défaut `output: 'static'`, `Astro.request.url` ne contient pas de paramètres de recherche, comme `?foo=bar`, car il n'est pas possible de les déterminer à l'avance lors des constructions statiques. Cependant, en mode `output: 'server'`, `Astro.request.url` contient les paramètres de recherche car ils peuvent être déterminés à partir d'une requête du serveur.
```
A closing `` ` `` was missing. Also, this is a code sample so no space before `:`.